zerver/tests/test_upload.py

```diff
@@ -1757,6 +1772,22 @@ def test_tarball_upload(self) -> None:
             path=os.path.basename(tarball_path))
         self.assertEqual(uri, expected_url)
 
+    @use_s3_backend
+    def test_delete_tarball(self) -> None:
+        create_s3_buckets(settings.S3_AVATAR_BUCKET)
+
+        user_profile = self.example_user("iago")
+        self.assertTrue(user_profile.is_realm_admin)
+
+        tarball_path = os.path.join(get_or_create_dev_uuid_var_path('test-backend'),
+                                    'tarball.tar.gz')
+        with open(tarball_path, 'w') as f:
+            f.write('dummy')
+
+        uri = upload_export_tarball(user_profile.realm, tarball_path)
+        self.assertIsNotNone(uri)
+        self.assertIsNone(delete_export_tarball('not_a_file'))
+        self.assertEqual(delete_export_tarball(uri), urllib.parse.urlparse(uri).path)
```

Comment:
I'm less sure about these, but I think it'd be better to extend test cases that already "upload" a tarball to do this deletion test, after checking the `get` logic is correct, rather than having an independent test function.  It'll eliminate a lot of semi-duplicated setup code, and again be more robust to changes in how we store things.

Reply:
Agreed, also updated the test name to be clear that we are testing both upload and deletion.